src/check/parse/tokenize/src/main.zig

```diff
@@ -0,0 +1,1399 @@
+        var escape: bool = false;
+        while (self.cursor.pos < self.cursor.buf.len) {
+            const c = self.cursor.buf[self.cursor.pos];
+            if (escape) {
+                switch (c) {
+                    '\\', '"', '\'', 'n', 'r', 't' => {
+                        escape = false;
+                        self.cursor.pos += 1;
+                    },
+                    'u' => {
+                        escape = false;
+                        self.cursor.pos += 1;
+                        self.cursor.require('(', MessageKind.InvalidUnicodeEscapeSequence);
+                        while (true) {
+                            if (self.cursor.peek() == ')') {
+                                self.cursor.pos += 1;
+                                break;
+                            } else if (self.cursor.peek() != null) {
+                                const next = self.cursor.peek() orelse 0;
+                                if ((next >= '0' and next <= '9') or
+                                    (next >= 'a' and next <= 'f') or
+                                    (next >= 'A' and next <= 'F'))
+                                {
+                                    self.cursor.pos += 1;
+                                } else {
+                                    self.cursor.pushMessageHere(.InvalidUnicodeEscapeSequence);
+                                    break;
+                                }
+                            } else {
+                                break;
+                            }
+                        }
+                    },
+                    else => {
+                        self.cursor.pushMessageHere(.InvalidEscapeSequence);
+                        escape = false;
+                        self.cursor.pos += 1;
+                    },
+                }
+            } else {
+                if (c == '\\') {
+                    escape = true;
+                    self.cursor.pos += 1;
+                } else if (c == '$' and self.cursor.peekAt(1) == '{') {
+                    self.cursor.pos += 1;
+                    var brace: BraceKind = undefined;
+                    if (term == '"') {
+                        if (multiline) {
+                            brace = .StringInterpolationMultiline;
+                        } else {
+                            brace = .StringInterpolation;
+                        }
+                    } else {
+                        if (multiline) {
+                            brace = .SingleQuoteInterpolationMultiline;
+                        } else {
+                            brace = .SingleQuoteInterpolation;
+                        }
+                    }
+                    try self.stack.append(brace);
+                    if (term == '"') {
+                        if (already_started) {
+                            return T.StringPart;
```

Comment:
In general I think you have a lot of inconsistent use of enum names. I think you should clean it up to always drop the enum name and just be `.StringPart`, etc.